modules/shared/src/main/scala/io/constellationnetwork/security/Hash.scala

```diff
@@ -24,12 +24,39 @@ object hash {
   }
 
   object Hash {
+    private val hexDigits = "0123456789abcdef".toCharArray
+    private val sha256 = MessageDigest.getInstance("SHA-256")
 
+    case class Sha256Digest(private val bytes: Array[Byte]) {
+      def toByteArray: Array[Byte] = Array.copyOf(bytes, bytes.length)
+
+      def toHexString: String =
+        bytes
+          .foldLeft(new StringBuilder(bytes.length * 2)) { (sb, b) =>
+            sb.append(hexDigits((b >> 4) & 0xf)).append(hexDigits(b & 0xf))
+          }
+          .toString
+
+      private[constellationnetwork] def toIndexedSeq: IndexedSeq[Byte] = bytes.toIndexedSeq
+    }
+
+    object Sha256Digest {
+      implicit val show: Show[Sha256Digest] = Show.show(s => s"Sha256Digest(${s.toHexString})")
+      implicit val eq: Eq[Sha256Digest] = Eq.instance((a, b) => a.bytes.sameElements(b.bytes))
```

Comment:
Instead of specifying custom Eq, you can derive existing one

Reply:
Using `@derive`? It doesn't compile. 

```
magnolia: could not find Eq.Typeclass for type Array[Byte]
```

But my implementation both compiles and works.

Reply:
It does not compile because you need to provide an implicit instance of an `Eq[Array[Byte]]` which is:
```scala
implicit val byteArrayEq: Eq[Array[Byte]] = Eq.instance(Arrays.equals(_, _))
```

and `Eq` instance for `Sha256Digest` you can simply derive. Many implementations would compile and work :)

Reply:
so you need an `Eq` instance for just `Array[Byte]` instead of an `Sha256Digest`. Many implementations would compile and work :)